src/k8s/pkg/k8sd/setup/containerd.go

```diff
@@ -16,119 +16,96 @@ import (
-	RegistryConfigDir string
-	PauseImage        string
-}
-
-type containerdConfig struct {
-	Version int                     `toml:"version"`
-	Plugins containerdConfigPlugins `toml:"plugins,omitempty"`
-}
-
-type containerdConfigPlugins struct {
-	CRI containerdConfigPluginsCRI `toml:"io.containerd.grpc.v1.cri,omitempty"`
-}
-
-type containerdConfigPluginsCRI struct {
-	Registry containerdConfigPluginsCRIRegistry `toml:"registry,omitempty"`
-}
-
-type containerdConfigPluginsCRIRegistry struct {
-	Configs map[string]containerdConfigPluginsCRIRegistryConfig `toml:"configs,omitempty"`
-}
-
-type containerdConfigPluginsCRIRegistryConfig struct {
-	Auth containerdConfigPluginsCRIRegistryConfigAuth `toml:"auth,omitempty"`
-}
+func defaultContainerdConfig(
+	cniConfDir string,
+	cniBinDir string,
+	importsDir string,
+	registryConfigDir string,
+	pauseImage string,
+) map[string]any {
+	return map[string]any{
+		"version":   2,
+		"oom_score": 0,
+		"imports":   []string{filepath.Join(importsDir, "*.toml")},
+
+		"grpc": map[string]any{
+			"uid":                   0,
+			"gid":                   0,
+			"max_recv_message_size": 16777216,
+			"max_send_message_size": 16777216,
+		},
 
-type containerdConfigPluginsCRIRegistryConfigAuth struct {
-	Username string `toml:"username,omitempty"`
-	Password string `toml:"password,omitempty"`
-	Token    string `toml:"token,omitempty"`
-}
+		"debug": map[string]any{
+			"uid":     0,
+			"gid":     0,
+			"address": "",
+			"level":   "",
+		},
 
-type containerdHostsConfig struct {
-	Server string                               `toml:"server,omitempty"`
-	Host   map[string]containerdHostsConfigHost `toml:"hosts,omitempty"`
-}
+		"metrics": map[string]any{
+			"address":        "",
+			"grpc_histogram": false,
+		},
 
-type containerdHostsConfigHost struct {
-	Capabilities []string `toml:"capabilities,omitempty"`
-	SkipVerify   bool     `toml:"skip_verify,omitempty"`
-	OverridePath bool     `toml:"override_path,omitempty"`
-}
+		"cgroup": map[string]any{
+			"path": "",
+		},
 
-func containerdAuthConfig(registries []types.ContainerdRegistry) containerdConfig {
-	authConfigs := make(map[string]containerdConfigPluginsCRIRegistryConfig, len(registries))
-	for _, registry := range registries {
-		if registry.Username != "" || registry.Password != "" || registry.Token != "" {
-			for _, url := range registry.URLs {
-				authConfigs[url] = containerdConfigPluginsCRIRegistryConfig{
-					Auth: containerdConfigPluginsCRIRegistryConfigAuth{
-						Username: registry.Username,
-						Password: registry.Password,
-						Token:    registry.Token,
+		"plugins": map[string]any{
+			"io.containerd.grpc.v1.cri": map[string]any{
+				"stream_server_address":       "127.0.0.1",
+				"stream_server_port":          "0",
+				"enable_selinux":              false,
+				"sandbox_image":               pauseImage,
+				"stats_collect_period":        10,
+				"enable_tls_streaming":        false,
+				"max_container_log_line_size": 16384,
```

Comment:
Do we want to set a higher line count by default, just in case other people are interested in this?